src/nodes/conv.rs

```diff
@@ -0,0 +1,384 @@
+    // To properly handle those cases, we should have our own ctime handling.
+    let ctime = Timespec { sec: attr.ctime(), nsec: attr.ctime_nsec() as i32 };
+
+    let perm = match attr.permissions().mode() {
+        // TODO(https://github.com/rust-lang/rust/issues/51577): Drop :: prefix.
+        mode if mode > ::std::u16::MAX as u32 => {
+            warn!("File system returned mode {} for {:?}, which is too large; set to 0400",
+                mode, path);
+            0o400
+        },
+        mode => (mode as u16) & !libc::S_IFMT,
+    };
+
+    let rdev = match attr.rdev() {
+        // TODO(https://github.com/rust-lang/rust/issues/51577): Drop :: prefix.
+        rdev if rdev > ::std::u32::MAX as u64 => {
+            warn!("File system returned rdev {} for {:?}, which is too large; set to 0",
+                rdev, path);
+            0
+        },
+        rdev => rdev as u32,
+    };
+
+    fuse::FileAttr {
+        ino: inode,
+        kind: filetype_fs_to_fuse(path, &attr.file_type()),
+        nlink: nlink,
+        size: len,
+        blocks: 0, // TODO(jmmv): Reevaluate what blocks should be.
+        atime: system_time_to_timespec(path, "atime", &attr.accessed()),
+        mtime: system_time_to_timespec(path, "mtime", &attr.modified()),
+        ctime: ctime,
+        crtime: system_time_to_timespec(path, "crtime", &attr.created()),
+        perm: perm,
+        uid: attr.uid(),
+        gid: attr.gid(),
+        rdev: rdev,
+        flags: 0,
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    use std::ffi::CString;
+    use std::fs::File;
+    use std::io::Write;
+    use std::os::unix;
+    use std::time::Duration;
+    use tempdir::TempDir;
+
+    #[test]
+    fn test_system_time_to_timespec_ok() {
+        let sys_time = SystemTime::UNIX_EPOCH + Duration::new(12345, 6789);
+        let timespec = system_time_to_timespec(
+            &Path::new("irrelevant"), "irrelevant", &Ok(sys_time));
+        assert_eq!(Timespec { sec: 12345, nsec: 6789 }, timespec);
+    }
+
+    #[test]
+    fn test_system_time_to_timespec_bad() {
+        let sys_time = SystemTime::UNIX_EPOCH - Duration::new(1, 0);
+        let timespec = system_time_to_timespec(
+            &Path::new("irrelevant"), "irrelevant", &Ok(sys_time));
+        assert_eq!(BAD_TIME, timespec);
+    }
+
+    #[test]
+    fn test_system_time_to_timespec_missing() {
+        let timespec = system_time_to_timespec(
+            &Path::new("irrelevant"), "irrelevant",
+            &Err(io::Error::from_raw_os_error(libc::ENOENT)));
+        assert_eq!(BAD_TIME, timespec);
+    }
+
+    /// Sets the permissions of a file to exactly those given, bypassing the umask.
+    fn chmod_exactly(path: &Path, perm: libc::mode_t) {
+        let path = path.as_os_str().to_str().unwrap().as_bytes();
+        let path = CString::new(path).unwrap();
+        assert_eq!(0, unsafe {
+            let old_umask = libc::umask(0);
```

Comment:
Uh, what? Since when does chmod pay any attention to umask?

Reply:
Uh, what? Exactly. I have no clue what I was thinking. (Well, I know what led me to this stupidity, but still.) Removed.